objects/screen.c

```diff
@@ -241,6 +243,143 @@ screens_exist(void)
     return globalconf.screens.len > 0;
 }
 
+#ifdef XCB_RANDR_GET_MONITORS
+static bool
+screen_scan_randr_monitors(void)
+{
+    xcb_randr_get_monitors_cookie_t monitors_c = xcb_randr_get_monitors(globalconf.connection, globalconf.screen->root, 1);
+    xcb_randr_get_monitors_reply_t *monitors_r = xcb_randr_get_monitors_reply(globalconf.connection, monitors_c, NULL);
+    xcb_randr_monitor_info_iterator_t monitor_iter;
+    bool found = false;
+
+    for(monitor_iter = xcb_randr_get_monitors_monitors_iterator(monitors_r); monitor_iter.rem; xcb_randr_monitor_info_next(&monitor_iter))
```

Comment:
The syntax requires long lines, as you see in this example. I don't have access to RandR 1.5 XCB headers right now, but I guess `for (int i = 0; i < xcb_randr_get_monitors_monitors_length(monitors_r); i++)` could work and be shorter (and then accessing that part of the reply as `xcb_randr_get_monitors_monitors(reply)[i]`.

However, feel free to do this how you want. Both approaches are equivalent and the one without iterators gets us longer lines elsewhere.